### PR TITLE
[ADD] hr_contract_usability

### PR DESCRIPTION
**Pull request by saltonmassally**
Makes several improvements to how contract is handled.
- Computes the first contract for an employee
- Computes the current contract forr an employee
- Set default trial period in HR configuration
- Set job of employee automatically to that of the lastest contract
  |

### CODE REVIEW

File: hr_contract_usability/models/hr_contract.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    
+    @api.onchange('trial_date_start')
+    @api.one
+    def onchange_trial_date_start(self):        
+        if self.trial_date_start:
+            res = self.env["ir.config_parameter"]\
+                      .get_param("hr_contract.default_trial_length")
+            res = res and int(res) or 60
+            end_dt = fields.Date.from_string(self.trial_date_start) \
+                                + relativedelta(days=res)
```
Comment (ghost) [inline hr_contract_usability/models/hr_contract.py:19]: try not using backslashes in python. This makes the code less readable. Use the parenthesis to break lines.

File: hr_contract_usability/models/hr_employee.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ 
+
+class hr_employee(models.Model):
+
+    _inherit = 'hr.employee'
+    
+    contract_id = fields.Many2one('hr.contract', 'urrent Contract', 
+                                  help='Latest contract of the employee',
+                                  store=True, readonly=True, index=True,
+                                  compute='_compute_current_contract')
```
Comment (ghost) [inline hr_contract_usability/models/hr_employee.py:13]: put one argument per line in your model fields

File: hr_contract_usability/models/hr_employee.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    job_id = fields.Many2one('hr.job', 'Job Title', readonly=True, store=True,
+                             related='contract_id.job_id')
+    
+    @api.one 
+    @api.depends('contract_ids', 'contract_ids.date_start', 
+                 'contract_ids.employee_id', 'contract_ids.job_id')
+    def _compute_first_contract(self):
+        if len(self.contract_ids):
+            contracts_sorted = self.contract_ids\
+                        .sorted(key=operator.itemgetter('date_start', 'id'))
```
Comment (ghost) [inline hr_contract_usability/models/hr_employee.py:26]: same thing, that kind of line break is very odd

File: hr_contract_usability/models/res_config.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
```
Comment (ghost) [inline hr_contract_usability/models/res_config.py:1]: put the licence in beginning of each file. This is advised by the GNU org.

**Comment by ghost** [comment]
Could we find a better name for this module ?

**Comment by saltonmassally** [comment]
@dufresnedavid I was thinking about breaking this module up into smaller addons, e.g. hr_contract_default_trial_length, etc

**Comment by dreispt** [comment]
You don't need a module to set defaults, that's a standard Odoo feature:
Just use the "Set Defaults" option in the debug/developer menu.

**Comment by saltonmassally** [comment]
Trail length should be configurable by users and we wouldn't want to give them access to debug also trail length date end is supposed to be relative to date_start

**Comment by saltonmassally** [comment]
Moved to 
https://github.com/OCA/hr/pull/161
https://github.com/OCA/hr/pull/160

**Comment by tarzan0820** [comment]
Contract Chronology bit moved to

**Pull request closed without merging**